tests/tensorflow/test_tensorflow2_autolog.py

```diff
@@ -42,6 +43,8 @@
 @pytest.fixture(autouse=True)
 def clear_session():
     yield
+    # Flush async logging to ensure we terminate thread pools.
+    mlflow.flush_async_logging()
```

Comment:
`flush` sets up a new thread. It'd be nice if we could avoid setting up a new thread.

https://github.com/mlflow/mlflow/blob/6b4d9e7063cca7498132de966bb695ec77ae8584/mlflow/utils/async_logging/async_artifacts_logging_queue.py#L74

Reply:
Yes, I can add a separate method for this